docs/demos/coast-to-coast-q4-2023.md

```diff
@@ -0,0 +1,950 @@
+>[!NOTE]
+>In this demo, we aren't going to make the Operator do anything.
+
+We will create a new Operator using the `operator-sdk`.
+
+- Create a new directory for the project and `cd` into it:
+```sh
+mkdir coastal && cd coastal
+```
+
+- Initialize the project:
+```sh
+operator-sdk init --domain coastal.operatorframework.io
+```
+
+- Create a new API:
+```sh
+operator-sdk create api \
+    --group coastal.operatorframework.io \
+    --version v1alpha1 \
+    --kind Coast \
+    --resource --controller
+```
+
+### Build the Operator and Bundle Images
+For this demo we are going to build a several different versions of this Operator:
+- `v1.0.0-alpha1`
+- `v1.0.0`
+- `v1.0.1`
+- `v1.1.0`
+- `v2.0.0`
+
+Initial setup:
+- Create the `manifests/` directory for building the plain bundle
+```sh
+  mkdir -p manifests
+```
+
+- Create a Dockerfile for the plain bundle image
+```sh
+cat << EOF > plainbundle.Dockerfile
+FROM scratch
+ADD manifests /manifests
+EOF
+```
+
+For each of the versions above, perform the following actions:
+- Generate the manifests
+```sh
+  make generate manifests
+```
+
+- Build the controller image
+```sh
+  make docker-build IMG="quay.io/operator-framework/coastal:{VERSION}"
+```
+
+- Populate the `manifests/` directory for building the plain bundle
+```sh
+  kustomize build config/default > manifests/manifests.yaml
+```
+
+- Build the plain bundle image
+```sh
+docker build -t quay.io/operator-framework/coastal-bundle:{VERSION} -f plainbundle.Dockerfile .
+```
+
+- Push the controller image and plain bundle image
+```sh
+docker push quay.io/operator-framework/coastal:{VERSION} && \
+docker push quay.io/operator-framework/coastal-bundle:{VERSION}
+```
+
+A handy little script to run all these steps for you:
+```sh
+#!/usr/bin/env bash
+set -e
+
+mkdir -p manifests
+
+cat << EOF > plainbundle.Dockerfile
+FROM scratch
+ADD manifests /manifests
+EOF
+
+versions=( v1.0.0-alpha1 v1.0.0 v1.0.1 v1.1.0 v2.0.0 )
+for version in "${versions[@]}"
+do
+  make generate manifests
+  make docker-build IMG="quay.io/operator-framework/coastal:${version}"
+  mkdir -p manifests
+  kustomize build config/default > manifests/manifests.yaml
+  docker build -t "quay.io/operator-framework/coastal-bundle:${version}" -f plainbundle.Dockerfile .
+  docker push "quay.io/operator-framework/coastal:${version}"
+  docker push "quay.io/operator-framework/coastal-bundle:${version}"
+done
+```
+
+### Create an initial File-Based Catalog (FBC) Image
+Now we need to build and push an initial File-Based Catalog (FBC) image containing
+our bundles. To help highlight the new polling functionality, we will generate a catalog containing
+only the `v1.0.0-alpha1` bundle and we will mimic active development by incrementally updating the catalog
+image to include more of our bundles.
+
+- Create a `catalog/` directory
+```sh
+mkdir catalog
+```
+
+- Create a Dockerfile for the catalog image
+```sh
+cat << EOF > catalog.Dockerfile
+FROM scratch
+ADD catalog /configs
```

Comment:
Can we add a directory structure (output of `tree .`) of top level catalog dir. Ran through the tutorial for building a catalog, and faced the error that catalog/ isn't available, because I had `index.yaml` inside parent dir. 

Just a nit, but would be helpful.

Reply:
Following this tutorial command by command should have resulted in a `tree` output of:
```sh
$ tree catalog/
catalog/
└── index.yaml

1 directory, 1 file
```

Following this tutorial without modifying the output paths in the commands should result in a successful build. While I can see how the output of this command would be helpful I'd prefer to do it as a follow up or provide a better explanation of building a File-Based Catalog image for OLMv1 in it's own document